### PR TITLE
Field title now uses custom axis title as default

### LINKED ISSUE #75: Tooltip not applying custom axis title
image

vl spec:

``` json
{
  "description": "A scatterplot showing horsepower and miles per gallons for various cars.",
  "data": {"url": "data/cars.json"},
  "mark": "circle",
  "encoding": {
    "x": {"field": "Horsepower","type": "quantitative", "axis": {"title": "HP"}},
    "y": {"field": "Miles_per_Gallon","type": "quantitative", "axis": {"title": "MPG"}}
  }
}
```

`vl.fieldDef.title(fieldDef, config);` returns `Horsepower`, `Miles_per_Gallon` instead of `HP`, `MPG`

### PR DESCRIPTION
**Pull request by sirahd**
- Fixes #75 and update examples to show that it uses axis title as default

### CODE REVIEW

File: exampleSpecs/scatter.json [hunk 1]
```diff
@@ -1,9 +1,9 @@
 {
-  "$schema": "https://vega.github.io/schema/vega-lite/v2.json",
```
Comment (domoritz) [inline exampleSpecs/scatter.json]: Why remove the schema?

Comment (sirahd) [reply exampleSpecs/scatter.json]: I just blindly copy and paste from issue. Will fix it.

File: src/supplementField.ts [hunk 2]
```diff
@@ -198,12 +198,14 @@ export function supplementFieldOption(fieldOption: FieldOption, fieldDef: FieldD
   if (!config.countTitle) {
     config.countTitle = vl.config.defaultConfig.countTitle; // use vl default countTitle
   }
-  supplementedFieldOption.title = fieldOption.title ?
-    fieldOption.title : vl.fieldDef.title(fieldDef, config);
+  let defaultTitle: string = vl.fieldDef.title(fieldDef, config);
+  if ((fieldDef as any).axis && (fieldDef as any).axis.title) {
```
Comment (domoritz) [inline src/supplementField.ts]: Why is the `as any` necessary?

File: src/supplementField.ts [hunk 3]
```diff
@@ -198,12 +198,14 @@ export function supplementFieldOption(fieldOption: FieldOption, fieldDef: FieldD
   if (!config.countTitle) {
     config.countTitle = vl.config.defaultConfig.countTitle; // use vl default countTitle
   }
-  supplementedFieldOption.title = fieldOption.title ?
-    fieldOption.title : vl.fieldDef.title(fieldDef, config);
+  let defaultTitle: string = vl.fieldDef.title(fieldDef, config);
```
Comment (domoritz) [inline src/supplementField.ts]: Isn't `string` the type already?

File: src/supplementField.ts (hunk 2 above)
Comment (domoritz) [reply src/supplementField.ts]: How can you use it if it's not a valid property? I think any is an indicator of an error in the logic.

Comment (kanitw) [reply src/supplementField.ts]: You can use a type guard for position fielddef (or add one if we dont have one yet)

**Comment by sirahd** [comment]
@domoritz Done!

File: src/supplementField.ts [hunk 4]
```diff
@@ -233,3 +243,19 @@ export function supplementFieldOption(fieldOption: FieldOption, fieldDef: FieldD
 
   return supplementedFieldOption;
 }
+
+/**
+ * Typeguard function for PositionFieldDef interface
+ */
+function isPositionFieldDef(fd: FieldDef<any>): fd is PositionFieldDef<any> {
+  // the fieldDef may still be a PositionFieldDef even if it doesn't have a legend
```
Comment (domoritz) [inline src/supplementField.ts]: axis

**Comment by domoritz** [comment]
Looks good to me. I defer to @kanitw for a review of the things I suggested.

**Pull request merged**